packages/core/strapi/src/commands/actions/plugin/link-watch/action.ts

```diff
@@ -0,0 +1,98 @@
+import boxen from 'boxen';
+import chalk from 'chalk';
+import concurrently from 'concurrently';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import nodemon from 'nodemon';
+import { outdent } from 'outdent';
+
+import { notifyExperimentalCommand } from '../../../utils/helpers';
+import { CLIContext } from '../../../types';
+import { loadPkg, validatePkg } from '../../../utils/pkg';
+
+interface ActionOptions {}
+
+export default async (_opts: ActionOptions, _cmd: unknown, { cwd, logger }: CLIContext) => {
+  try {
+    /**
+     * Notify users this is an experimental command.
+     */
+    await notifyExperimentalCommand('plugin:watch:link', { force: true });
+
+    const outDir = './dist';
+    const extensions = 'ts,js,png,svg,gif,jpeg,css';
+
+    nodemon({
+      watch: [outDir],
+      ext: extensions,
+      exec: 'yalc push --changed',
+    });
```

Comment:
node does support running it in watch mode starting from version 18, example:
`node --watch main.js`

Edit: Nvm, it looks that api doesnt support wathing specific extensions like you need here